### ISSUE TITLE
Display compatibility issue with company-mode

### ISSUE THREAD

**Issue by john2x** [alpaker/fill-column-indicator#54]
When `fci-mode` is enabled and a `company-mode` auto-complete popup shows up, the first completion item is moved beyond the ruler. See screenshot.

![__projects_clojure_ledgerview_src_ledgerview_ledger_core_clj](https://cloud.githubusercontent.com/assets/332885/4101410/b0b52a8e-30e8-11e4-82fc-a169fb73b424.png)

**Comment by hunterboerner** [comment]
Status update?

**Comment by john2x** [comment]
I've encountered another display issue with fci-mode + hideshow-mode. That has prompted me to stop using fci-mode for the moment as there are too many incompatibilities with other modes I'm using (and likely other modes I might use in the future).

Maybe a wiki/FAQ entry for compatibility issues would be useful?

**Comment by root42** [comment]
Any news on this one? I am using Emacs24 with the usual MELPA packages, and company + fci-mode still don't play nice together.

**Comment by eigengrau** [comment]
Dupe of #21.

**Comment by sukiida** [comment]
@root42 [https://github.com/alpaker/Fill-Column-Indicator/issues/21](21) has a discussion.
I translate purcell's workaround into new form of advice and it works for me with emacs24.5.1 and company mode.
my code is like this:
```elisp
(defun on-off-fci-before-company(command)
  (when (string= "show" command)
    (turn-off-fci-mode))
  (when (string= "hide" command)
    (turn-on-fci-mode)))

(advice-add 'company-call-frontends :before #'on-off-fci-before-company)
```
>  purcell commented on 13 Jul 2012

>I just wanted to note, particularly for the benefit of @alpaker, that you can work around this problem by temporarily suspending fci-mode while popups are being displayed:

>(defvar sanityinc/fci-mode-suppressed nil)
>(defadvice popup-create (before suppress-fci-mode activate)
> ."Suspend fci-mode while popups are visible"
>  (set (make-local-variable 'sanityinc/fci-mode-suppressed) fci-mode)
>  (when fci-mode
>    (turn-off-fci-mode)))
>(defadvice popup-delete (after restore-fci-mode activate)
>  "Restore fci-mode when all popups have closed"
>  (when (and (not popup-instances) sanityinc/fci-mode-suppressed)
>    (setq sanityinc/fci-mode-suppressed nil)
>    (turn-on-fci-mode)))

**Comment by alpaker** [comment]
Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

**Comment by posita** [comment]
> Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

Thanks @alpaker for the heads up. Please consider indicating status and inviting new maintainers on https://www.emacswiki.org/emacs/FillColumnIndicator as well.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.

**Comment by alpaker** [comment]
I've  updated the emacswiki page as well.
